lib/pusher/client.rb

```diff
@@ -88,6 +105,12 @@ def encrypted?
       @scheme == 'https'
     end
 
+    def cluster=(cluster)
```

Comment:
If the user explicitly calls `pusher.cluster = 'eu'` then we shouldn't ignore that. We should reset all the other conflicting settings.